loom/graphics/gfxQuadRenderer.cpp

```diff
@@ -203,39 +176,42 @@ void QuadRenderer::submit()
                     default:
                         lmAssert(false, "Unsupported smoothing: %d", tinfo.smoothing);
                 }
+                
+                sTextureStateValid = true;
+            }
 
+            if (!sBlendStateValid)
+            {
                 // Blend mode.
                 Graphics::context()->glEnable(GL_BLEND);
                 Graphics::context()->glBlendFuncSeparate(sSrcBlend, sDstBlend, sSrcBlend, sDstBlend);
 
-                Graphics_SetCurrentGLState(GFX_OPENGL_STATE_QUAD);
+                sBlendStateValid = true;
             }
+            
+            Graphics_SetCurrentGLState(GFX_OPENGL_STATE_QUAD);
+            
+            // Setting the buffer to null supposedly enables better performance because it enables the driver to do some optimizations.
+            Graphics::context()->glBufferData(GL_ARRAY_BUFFER, batchedVertexCount*sizeof(VertexPosColorTex), NULL, GL_STREAM_DRAW);
+            Graphics::context()->glBufferData(GL_ARRAY_BUFFER, batchedVertexCount*sizeof(VertexPosColorTex), batchedVertices, GL_STREAM_DRAW);
 
             // And bind indices and draw.
-            Graphics::context()->glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, sIndexBufferHandle);
+            Graphics::context()->glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, indexBufferId);
             Graphics::context()->glDrawElements(GL_TRIANGLES,

```

Comment:
I'd love to have this be - upload once then draw, draw, draw. But you tried this and had bad results?